web/ui/react-app/src/pages/rules/RulePanel.tsx

```diff
@@ -0,0 +1,50 @@
+import React, { FC, CSSProperties, Fragment } from 'react';
+import { Rule } from '../../types/types';
+import { Link } from '@reach/router';
+import { createExpressionLink, mapObjEntries, formatRange } from '../../utils';
+
+interface RulePanelProps {
+  rule: Rule;
+  styles?: CSSProperties;
+  tag?: keyof JSX.IntrinsicElements;
+}
+
+export const RulePanel: FC<RulePanelProps> = ({ rule, styles = {}, tag }) => {
+  const Tag = tag || Fragment;
+  const { name, query, labels } = rule;
+  const style = { background: '#f5f5f5', ...styles };
+  return (
+    <Tag {...(tag ? { style } : {})}>
+      <pre className="m-0" style={tag ? undefined : style}>
+        <code>
+          {rule.type === 'alerting' ? 'alert: ' : 'record: '}
+          <Link to={createExpressionLink(`ALERTS{alertname="${name.replace(/"/g, '\\"')}"}`)}>{name}</Link>
```

Comment:
A recording rule does not produce `ALERTS` series, it records directly into the name given after `record:`, so for recording rules you will want to point the expression link directly at that metric name, without the `ALERTS{alertname="..."}` around it.